packages/flutter/lib/src/foundation/diagnostics.dart

```diff
@@ -2771,12 +2771,12 @@ class DiagnosticsProperty<T> extends DiagnosticsNode {
     _valueComputed = true;
     assert(_computeValue != null);
     try {
-      _value = _computeValue!();
+      _value = WeakValue<T>(_computeValue!(), keepAlive: true);
```

Comment:
we should audit all the _computeValue closures in Flutter. Some might be keeping objects alive rather than using WeakRef.